src/studio-home/tabs-section/courses-tab/index.tsx

```diff
@@ -1,67 +1,179 @@
 import React from 'react';
 import { useLocation } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
-import { useIntl } from '@edx/frontend-platform/i18n';
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
 import {
   Icon,
   Row,
   Pagination,
   Alert,
   Button,
+  Form,
 } from '@openedx/paragon';
 import { Error } from '@openedx/paragon/icons';
 
 import { COURSE_CREATOR_STATES } from '@src/constants';
-import { getStudioHomeData, getStudioHomeCoursesParams } from '@src/studio-home/data/selectors';
+import { getStudioHomeData, getStudioHomeCoursesParams, getLoadingStatuses } from '@src/studio-home/data/selectors';
 import { resetStudioHomeCoursesCustomParams, updateStudioHomeCoursesCustomParams } from '@src/studio-home/data/slice';
 import { fetchStudioHomeData } from '@src/studio-home/data/thunks';
-import CardItem from '@src/studio-home/card-item';
+import { CardItem } from '@src/studio-home/card-item';
 import CollapsibleStateWithAction from '@src/studio-home/collapsible-state-with-action';
 import ProcessingCourses from '@src/studio-home/processing-courses';
 import { LoadingSpinner } from '@src/generic/Loading';
 import AlertMessage from '@src/generic/alert-message';
+import { RequestStatus } from '@src/data/constants';
 import messages from '../messages';
 import CoursesFilters from './courses-filters';
 import ContactAdministrator from './contact-administrator';
 import './index.scss';
 
-interface Props {
-  coursesDataItems: {
-    courseKey: string;
-    displayName: string;
-    lmsLink: string | null;
-    number: string;
-    org: string;
-    rerunLink: string | null;
-    run: string;
-    url: string;
-  }[];
-  showNewCourseContainer: boolean;
-  onClickNewCourse: () => void;
-  isShowProcessing: boolean;
+export interface MigrationStatusProps {
+  courseId: string;
+  allVisibleCourseIds: string[];
+}
+
+interface CardListProps {
+  currentPage: number;
+  handlePageSelected: (page: any) => void;
+  handleCleanFilters: () => void;
+  onClickCard?: (courseId: string) => void;
   isLoading: boolean;
-  isFailed: boolean;
-  numPages: number;
-  coursesCount: number;
+  isFiltered: boolean;
+  hasAbilityToCreateCourse?: boolean;
+  showNewCourseContainer?: boolean;
+  onClickNewCourse?: () => void;
+  inSelectMode?: boolean;
+  selectedCourseId?: string;
+  migrationStatusWidget?: React.ComponentType<MigrationStatusProps> | null;
```

Comment:
```suggestion
  migrationStatusWidget?: React.ComponentType<MigrationStatusProps>;
```
Nit: If it's optional with `?` it already accepts `undefined` so I don't think you also need to specify `null`.